src/test/java/co/rsk/bitcoinj/script/ScriptTest.java

```diff
@@ -502,54 +509,140 @@ public void getNumberOfSignaturesRequiredToSpend_fast_bridge_erp_redeem_script()
+            federationKeys,
+            erpFederationKeys,
             500L
         );
 
         testGetSigInsertionIndex(erpRedeemScript);
     }
 
+    @Test
+    public void getSigInsertionIndexWhenP2shRedeemScriptShouldReturnIndex() {
+        Script redeemScript = RedeemScriptUtils.createP2shErpRedeemScript(
+            federationKeys,
+            erpFederationKeys,
+            600L
+        );
+
+        testGetSigInsertionIndex(redeemScript);
+    }
+
     @Test
     public void getSigInsertionIndex_fast_bridge_erp_redeem_script() {
         Script fastBridgeErpRedeemScript = RedeemScriptUtils.createFastBridgeErpRedeemScript(
-            btcECKeyList,
-            erpFedECKeyList,
+            federationKeys,
+            erpFederationKeys,
             500L,
             Sha256Hash.of(new byte[]{1}).getBytes()
         );
 
         testGetSigInsertionIndex(fastBridgeErpRedeemScript);
     }
 
+    @Test
+    public void getSigInsertionIndexWhenFlyoverP2shRedeemScriptShouldReturnIndex() {
+        Script flyoverP2shRedeemScript = RedeemScriptUtils.createFastBridgeP2shErpRedeemScript(
+            federationKeys,
+            erpFederationKeys,
+            500L,
+            Sha256Hash.of(new byte[]{1}).getBytes()
+        );
+
+        testGetSigInsertionIndex(flyoverP2shRedeemScript);
+    }
+
     @Test
     public void getSigInsertionIndex_no_fast_bridge_redeem_script() {
-        Script redeemScript = RedeemScriptUtils.createStandardRedeemScript(btcECKeyList);
+        Script redeemScript = RedeemScriptUtils.createStandardRedeemScript(federationKeys);
         testGetSigInsertionIndex(redeemScript);
     }
 
+    @Test
+    public void getSigInsertionIndexWhenEmptyScriptShouldThrowScriptException() {
+        Script redeemScript = new ScriptBuilder().build();
+        Sha256Hash hashForSignature = Sha256Hash.of(new byte[]{1});
+        BtcECKey signingKey = BtcECKey.fromPrivate(BigInteger.valueOf(800));
+
+        try {
+            redeemScript.getSigInsertionIndex(hashForSignature, signingKey);
+            Assert.fail();
+        } catch (ScriptException e) {
+            Assert.assertEquals("Script does not match ScriptSig format.", e.getMessage());
+        }
+    }
+
+    @Test
+    public void getSigInsertionIndexWhenP2shScriptShouldThrowNullPointerException() {
+        Script p2shOutputScript = ScriptBuilder.createP2SHOutputScript(
+            2,
+            Arrays.asList(fedKey1, fedKey2, fedKey3)
+        );
+
+        Sha256Hash hashForSignature = Sha256Hash.of(new byte[]{1});
+        BtcECKey signingKey = BtcECKey.fromPrivate(BigInteger.valueOf(800));
+
+        try {
+            p2shOutputScript.getSigInsertionIndex(hashForSignature, signingKey);
+            Assert.fail();
+        } catch (Exception e) {
+            Assert.assertTrue(e instanceof NullPointerException);
+        }
+    }
+
+    @Test
+    public void getSigInsertionIndexWhenTestnetHardcodeRedeemScriptThrowNullPointerException() {
```

Comment:
NPE here? Shouldn't this script be parsed as NoRedeemScriptParser?

Reply:
Solved. Now, it throws a ScriptException when `NoScriptSig` tries to `getSigInsertionIndex`.